### DISCUSSION TITLE
Represent class hierarchy

### DISCUSSION THREAD

**Discussion by mkuzmin** [terrastruct/d2#2025] [Q&A]
I'm trying to draw relations between objects and their properties .
Is there a way to make this chart more compact?
![kubernetes-compute](https://github.com/user-attachments/assets/1cad085c-e375-4028-b26c-83ca23a3cf6e)

It looks neither Class nor SQL Table support inner classes. And the Class diagram doesn't allow to create a connection from specific property.

**Comment by alixander** [comment]
What's the D2 file for this?
